sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/RowEncoder.scala

```diff
@@ -214,6 +215,7 @@ object RowEncoder {
       } else {
         nonNullOutput
       }
+    case _ => inputObject
```

Comment:
can we add a comment? `For other data types, just return the internal catalyst value as it is.`

Reply:
Done